app/services/recaptcha_validator.rb

```diff
@@ -0,0 +1,59 @@
+class RecaptchaValidator
+  VERIFICATION_ENDPOINT = 'https://www.google.com/recaptcha/api/siteverify'.freeze
+
+  class ValidationError < StandardError; end
+
+  attr_reader :score_threshold, :analytics
+
+  def initialize(score_threshold: 0.0, analytics: nil)
+    @score_threshold = score_threshold
+    @analytics = analytics
+  end
+
+  def exempt?
+    !score_threshold.positive?
+  end
+
+  def valid?(recaptcha_token)
+    return true if exempt?
+    return false if recaptcha_token.blank?
+
+    response = faraday.post(
+      VERIFICATION_ENDPOINT,
+      URI.encode_www_form(
+        secret: IdentityConfig.store.recaptcha_secret_key,
+        response: recaptcha_token,
+      ),
+    ) do |request|
+      request.options.context = { service_name: 'recaptcha' }
+    end
+
+    log_analytics(response.body)
+
+    if !response.body['success']
+      raise ValidationError.new("reCAPTCHA validation error: #{response.body['error-codes']}")
+    end
+
+    response.body['score'] >= score_threshold
+  rescue Faraday::Error, ValidationError => error
+    NewRelic::Agent.notice_error(error)
```

Comment:
In the past, when we've manually reported errors to NR, we worry because we think these are user-facing exceptions. WDYT of logging a warning or something else we can check for?

Reply:
> In the past, when we've manually reported errors to NR, we worry because we think these are user-facing exceptions. WDYT of logging a warning or something else we can check for?

Technically, we already are logging something which we could track for this ([related spec](https://github.com/18F/identity-idp/blob/1c0634cb1d12416331fcc04440a529d495aaf7cd/spec/services/recaptcha_validator_spec.rb#L67-L79)), so I could remove this altogether?

The thinking was that an unsuccessful result from reCAPTCHA is very likely to be due to a misconfiguration on our end ([see error code reference](https://developers.google.com/recaptcha/docs/verify#error_code_reference)) and ought never happen, so we'd want to have some urgency in being made aware of it.

Reply:
Remove NewRelic notice in 6574aeaf5.